generic_relations/tests/test_serializers.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import unicode_literals
+
+from django.test import TestCase
+
+from rest_framework import serializers
+
+from generic_relations.serializers import GenericModelSerializer
+from generic_relations.tests.models import Bookmark, Note
+
+from .test_relations import BookmarkSerializer, NoteSerializer
+
+
+class TestGenericModelSerializer(TestCase):
+    def setUp(self):
+        self.bookmark = Bookmark.objects.create(
+            url='https://www.djangoproject.com/')
+        self.note = Note.objects.create(text='Remember the milk')
+        self.note2 = Note.objects.create(text='Reticulate the splines')
+
+        self.serializer = GenericModelSerializer(
+            {
+                Bookmark: BookmarkSerializer(),
+                Note: NoteSerializer(),
+            }
+        )
+        self.list_serializer = serializers.ListSerializer(child=self.serializer)
+
+    def test_serialize(self):
+        self.assertEqual(
+            self.serializer.to_representation(self.bookmark),
+            {'url': 'https://www.djangoproject.com/'},
+        )
+        self.assertEqual(
+            self.serializer.to_representation(self.note),
+            {'text': 'Remember the milk'},
+        )
+
+    def test_deserialize(self):
+        self.assertEqual(
+            self.serializer.to_internal_value({'url': 'https://www.djangoproject.com/'}),
+            {'url': 'https://www.djangoproject.com/'},
+        )
+        self.assertEqual(
+            self.serializer.to_internal_value({'text': 'Remember the milk'}),
+            {'text': 'Remember the milk'},
```

Comment:
After reviewing, I think it's correct. `RelatedField` resolves to a model because that's what it does, but serializers provide a dict which can be used to update a model.

At a higher level though, GenericModelSerializer isn't responsible for this - the serializers we delegate to are. If they return a model that's up to them :)

I suppose the question becomes whether GenericModelSerializer should override `create` and `update` and try to delegate them to its' child serializers. Probably the answer is yes.